1.0/en/0x10-C04-Infrastructure.md

```diff
@@ -2,110 +2,324 @@
+| **4.16.1** | **Verify that** sensitive AI models execute within Intel SGX enclaves, AMD SEV-SNP, or ARM TrustZone with encrypted memory and attestation verification. | 3 | D/V |
+| **4.16.2** | **Verify that** confidential containers (Kata Containers, gVisor with confidential computing) isolate AI workloads with hardware-enforced memory encryption. | 3 | D/V |
+| **4.16.3** | **Verify that** remote attestation validates enclave integrity before loading AI models with cryptographic proof of execution environment authenticity. | 3 | D/V |
+| **4.16.4** | **Verify that** confidential AI inference services prevent model extraction through encrypted computation with sealed model weights and protected execution. | 3 | D/V |
+| **4.16.5** | **Verify that** trusted execution environment orchestration manages secure enclave lifecycle with remote attestation and encrypted communication channels. | 3 | D/V |
+| **4.16.6** | **Verify that** secure multi-party computation (SMPC) enables collaborative AI training without exposing individual datasets or model parameters. | 3 | D/V |
 
 ---
 
-## C4.4 Secrets Management & Environment Hardening
+## C4.17 Zero-Knowledge Infrastructure
 
-Compromised API keys remain a top breach vector. Hardware‑backed storage and rigorous rotation shrink the window of exposure.
+Implement zero-knowledge proof systems for privacy-preserving AI verification and authentication without revealing sensitive information.
 
 | # | Description | Level | Role |
 |:--------:|--------------------------------------------------------------------------------------------|:---:|:---:|
-| **4.4.1** | **Verify that** all secrets are rotated at least every 90 days or immediately upon personnel change. | 1 | D/V |
-| **4.4.2** | **Verify that** encryption keys are stored in TPM, HSM, or cloud KMS with automatic rotation and audit logging enabled. | 1 | D/V |
-| **4.4.3** | **Verify that** container images and start‑up scripts audit environment variables for sensitive data and block builds on leakage. | 2 | D |
-| **4.4.4** | **Verify that** SSH access to production nodes requires MFA and is disabled for service accounts. | 2 | V |
+| **4.17.1** | **Verify that** zero-knowledge proofs (ZK-SNARKs, ZK-STARKs) verify AI model integrity and training provenance without exposing model weights or training data. | 3 | D/V |
+| **4.17.2** | **Verify that** ZK-based authentication systems enable privacy-preserving user verification for AI services without revealing identity information. | 3 | D/V |
+| **4.17.3** | **Verify that** private set intersection (PSI) protocols enable secure data matching for federated AI without exposing individual datasets. | 3 | D/V |
+| **4.17.4** | **Verify that** zero-knowledge machine learning (ZKML) systems enable verifiable AI inference with cryptographic proof of correct computation. | 3 | D/V |
```

Comment:
New to this project, so apologies if this question is off base. Do we want to include glossary links defining or explaining concepts like this and some of the other acronyms throughout the PR?